compiler/rustc_mir/src/transform/instcombine.rs

```diff
@@ -153,15 +261,26 @@ impl Visitor<'tcx> for OptimizationFinder<'b, 'tcx> {
             }
         }
 
+        let _ = self.find_eq_not(rvalue, location);
+
         self.find_unneeded_equality_comparison(rvalue, location);
 
         self.super_rvalue(rvalue, location)
     }
 }
 
+#[derive(Clone)]
+struct EqNotOptInfo<'tcx> {
+    op1: Operand<'tcx>,
+    op2: Operand<'tcx>,
```

Comment:
I think it would good to add some doc comments explaining what these fields are for. These two in particular are fairly opaque names but it would be good to have comments for the other fields too.